### ISSUE TITLE
`maxDecodeStepCount` 400 Error while using `GoogleVertexAI("gemini-1.5-pro")` API

### ISSUE THREAD

**Issue by imflash217** [stanfordnlp/dspy#1430]
I am using a simple CoT module for binary classification and "reasoning" task with "Gemini-1.5-pro" model API. But I am constantly getting the following error and it produces some weird responses.

```
Going to sample between 1 and 4 traces per predictor.
Will attempt to bootstrap 16 candidate sets.
  0%|                                                                                                                                                                                             | 0/11 [00:00<?, ?it/s]←[2m2024-08-27T00:01:25.684723Z←[0m [←[31m←[1merror    ←[0m] ←[1mError for example in dev set:            400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again.←[0m [←[0m←[1m←[34mdspy.evaluate.evaluate←[0m]←[0m ←[36mfilename←[0m=←[35mevaluate.py←[0m ←[36mlineno←[0m=←[35m183←[0m
Average Metric: 2.0 / 7  (28.6):  64%|██████████████████████████████████████████████████████████████████████████████████████████████▏                                                     | 7/11 [00:03<00:01,  2.23it/s]←[2m2024-08-27T00:01:27.289144Z←[0m [←[31m←[1merror    ←[0m] ←[1mError for example in dev set:            400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again.←[0m [←[0m←[1m←[34mdspy.evaluate.evaluate←[0m]←[0m ←[36mfilename←[0m=←[35mevaluate.py←[0m ←[36mlineno←[0m=←[35m183←[0m
Average Metric: 3.0 / 11  (27.3): 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11/11 [00:03<00:00,  2.82it/s]
Score: 27.27 for set: [0]
New best sscore: 27.27 for seed -3
Scores so far: [27.27]
Best score: 27.27
  0%|                                                                                                                                                                                             | 0/11 [00:00<?, ?it/s]←[2m2024-08-27T00:01:28.963632Z←[0m [←[31m←[1merror    ←[0m] ←[1mError for example in dev set:            400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again.←[0m [←[0m←[1m←[34mdspy.evaluate.evaluate←[0m]←[0m ←[36mfilename←[0m=←[35mevaluate.py←[0m ←[36mlineno←[0m=←[35m183←[0m
Average Metric: 2.0 / 7  (28.6):  64%|██████████████████████████████████████████████████████████████████████████████████████████████▏                                                     | 7/11 [00:02<00:01,  2.50it/s]←[2m2024-08-27T00:01:30.433741Z←[0m [←[31m←[1merror    ←[0m] ←[1mError for example in dev set:            400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again.←[0m [←[0m←[1m←[34mdspy.evaluate.evaluate←[0m]←[0m ←[36mfilename←[0m=←[35mevaluate.py←[0m ←[36mlineno←[0m=←[35m183←[0m
Average Metric: 3.0 / 11  (27.3): 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11/11 [00:03<00:00,  3.46it/s]
Score: 27.27 for set: [11]
Scores so far: [27.27, 27.27]
Best score: 27.27
36%|█████████████████████████████████████████████████████████████████▊                                                                                                                   | 4/11 [00:05<00:10,  1.44s/it]←[2m2024-08-27T00:01:37.755313Z←[0m [←[31m←[1merror    ←[0m] ←[1mFailed to run or to evaluate example <flashrag.core.dataloader.CustomExample object at 0x000001688656B590> with <function answer_f1_match_01 at 0x00000168BCC02020> due to 400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again..←[0m [←[0m←[1m←[34mdspy.teleprompt.bootstrap←[0m]←[0m ←[36mfilename←[0m=←[35mbootstrap.py←[0m ←[36mlineno←[0m=←[35m211←[0m
82%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████                                 | 9/11 [00:13<00:03,  1.58s/it]←[2m2024-08-27T00:01:45.302402Z←[0m [←[31m←[1merror    ←[0m] ←[1mFailed to run or to evaluate example <flashrag.core.dataloader.CustomExample object at 0x00000168BCCCBDD0> with <function answer_f1_match_01 at 0x00000168BCC02020> due to 400 Unable to submit request because it has a maxDecodeStepCount value of 10000 but the supported range is from 1 (inclusive) to 2049 (exclusive). Update the value and try again..←[0m [←[0m←[1m←[34mdspy.teleprompt.bootstrap←[0m]←[0m ←[36mfilename←[0m=←[35mbootstrap.py←[0m ←[36mlineno←[0m=←[35m211←[0m
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11/11 [00:15<00:00,  1.42s/it]
Bootstrapped 3 full traces after 11 examples in round 0.
Average Metric: 6 / 11  (54.5): 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11/11 [00:05<00:00,  1.86it/s]
Score: 54.55 for set: [11]
New best sscore: 54.55 for seed -1
Scores so far: [27.27, 27.27, 54.55]
Best score: 54.55
Average of max per entry across top 1 scores: 0.5454545454545454
Average of max per entry across top 2 scores: 0.5454545454545454
Average of max per entry across top 3 scores: 0.5454545454545454
Average of max per entry across top 5 scores: 0.5454545454545454
Average of max per entry across top 8 scores: 0.5454545454545454
Average of max per entry across top 9999 scores: 0.5454545454545454
```

I could not find `maxDecodeStepCount` as a parameter for GoogleVertexAI or "gemini-1.5-pro" model APIs.

Any help.

Thanks

**Comment by arnavsinghvi11** [comment]
Hi @imflash217 , 

feel free to add this to the [GoogleVertexAI](https://github.com/stanfordnlp/dspy/blob/bdf52ca21175f0aa0ca9e9ebc54fe1d7d059a06f/dsp/modules/google_vertex_ai.py#L35) client as needed via PR. I'm not aware of the interface, but maybe this parameter can be included in the client kwargs?

**Comment by okhat** [comment]
Thanks for opening this! We released DSPy 2.5 yesterday. I think the new `dspy.LM` and the underlying `dspy.ChatAdapter` will probably resolve this problem.

Here's the (very short) migration guide, it should typically take you 2-3 minutes to change the LM definition and you should be good to go: https://github.com/stanfordnlp/dspy/blob/main/examples/migration.ipynb

Please let us know if this resolves your issue. I will close for now but please feel free to re-open if the problem persists.
